sklearn/kernel_approximation.py

```diff
@@ -360,6 +363,12 @@ def fit(self, X, y=None):
         )
 
         self.random_offset_ = random_state.uniform(0, 2 * np.pi, size=self.n_components)
+
```

Comment:
usually in `_validate_data`, we will add the following:

```python
dtype=[np.float64, np.float32]
```

Reply:
The same option should be added in `transform`.

Reply:
For `fit`, I think this ends up to be a memory regression if `X` is an int array.

For `transform`, I'm okay with the casting. I think `safe_sparse_dot` ends up doing it anyways.

Reply:
I agree with @thomasjpfan that it should only be done in tranform and not in fit

Reply:
Given the comment in https://github.com/scikit-learn/scikit-learn/pull/24317#discussion_r964653974 (my conclusion is "The PR is fine as it is") I am not sure what the take away from this comment thread and from https://github.com/scikit-learn/scikit-learn/pull/24317#discussion_r963019800 is.